scripts/globals/job_utils/rune_fencer.lua

```diff
@@ -50,8 +50,163 @@ local function enforceRuneCounts(target)
+    for _, effect in ipairs(effects) do
+        local type = effect:getType()
+        if type >= xi.effect.IGNIS and type <= xi.effect.TENEBRAE then
+            HPHealAmount = HPHealAmount + getRuneHealAmount(effect, target)
+        elseif
+            type == xi.effect.POISON or type == xi.effect.PARALYSIS or
+            type == xi.effect.BLINDNESS or type == xi.effect.SILENCE or type == xi.effect.MUTE or type == xi.effect.CURSE_I or
+            type == xi.effect.CURSE_II or type == xi.effect.DOOM or type == xi.effect.VIRUS or
+            type == xi.effect.PLAGUE or type == xi.effect.PETRIFICATION
+        then
+            debuffs[debuffCount+1] = type
+        debuffCount = debuffCount + 1
+        end
+
+        if type == xi.effect.TENEBRAE then -- runes that also restore MP
+            tenebraeRuneCount = tenebraeRuneCount + 1
+        end
+    end
+
+    if tenebraeRuneCount > 0 then -- only restore MP if there's one or more tenebrae rune active
+        local MPHealAmount = math.floor(divineMagicSkillLevel / 10 * (100 + target:getJobPointLevel(xi.jp.VIVACIOUS_PULSE_EFFECT)) / 100) * (tenebraeRuneCount + 1)
+        target:addMP(MPHealAmount) -- augment bonusPct does not apply here according to testing.
+    end
+
+    if debuffCount > 0 and (target:getEquipID(xi.slot.HEAD) == 26782 or target:getEquipID(xi.slot.HEAD) == 26783) then -- add random removal of Poison, Paralyze, Blind, Silence, Mute, Curse, Bane, Doom, Virus, Plague, Petrification via AF3 head (source: https://www.bg-wiki.com/ffxi/Erilaz_Galea)
+        target:delStatusEffect(debuffs[math.random(debuffCount)])
+    end
+
+    HPHealAmount = HPHealAmount * bonusPct
+    if (target:getHP() + HPHealAmount > target:getMaxHP()) then
+        HPHealAmount = target:getMaxHP() - target:getHP() -- don't go over cap
+    end
+
+    return HPHealAmount
+end
 
 xi.job_utils.rune_fencer.useRuneEnchantment = function(player, target, ability, effect)
     enforceRuneCounts(target)
     applyRuneEnhancement(effect, target)
 end
+
+xi.job_utils.rune_fencer.useSwordplay = function(player, target, ability)
+    local power = 0 --Swordplay starts at +0 bonus without swaps
+
+    -- see https://www.bg-wiki.com/ffxi/Sleight_of_Sword for levels
+    local meritBonus =  player:getMerit(xi.merit.MERIT_SLEIGHT_OF_SWORD)
+    local augBonus = 0 -- augBonus = 2 per level of merit
+
+    -- gear bonuses from https://www.bg-wiki.com/ffxi/Swordplay
+    if player:getMainJob() == xi.job.RUN and target:getMainLvl() == 99 then -- don't bother with gear boost checks until 99 and main RUN
+        local tickPower = 3 -- Tick power appears to be 3/tick, not 6/tick if RUN main and 3/tick if RUN sub; source : https://www.ffxiah.com/forum/topic/37086/endeavoring-to-awaken-a-guide-to-rune-fencer/180/#3615377
+        local handsSlotID = target:getEquipID(xi.slot.HANDS)
+
+        -- add starting tick bonuses if appropriate gear is equipped
+        if handsSlotID      == 27018 then
```

Comment:
More things that could go into items.lua

Reply:
> More things that could go into items.lua

nay, mods mods mods

Reply:
I'm punting magic number changes until we get the appropriate item mods into the main branch. It will get fixed in a future PR, and I will note it in the progress chart. They will exist here as reminders for _what_ items need to have their sql entries updated. Don't need to duplicate work especially when we know the better way is on the horizon.